core/shared/src/main/scala/spire/algebra/Homomorphism.scala

```diff
@@ -0,0 +1,54 @@
+package spire.algebra
+
+import java.math.BigInteger
+
+import spire.math.{Complex, SafeLong}
+
+trait Homomorphism[-A, +B, +F[_]] extends Function1[A, B]
```

Comment:
I don't think this should extend `Function1`, since it means imported homomorphisms will also add implicit conversions which may be unexpected.